api/webhook.go

```diff
@@ -356,14 +359,27 @@ func incomingWebhook(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	r.ParseForm()
 
-	var parsedRequest *model.IncomingWebhookRequest
+	var payload io.Reader
 	contentType := r.Header.Get("Content-Type")
 	if strings.Split(contentType, "; ")[0] == "application/json" {
-		parsedRequest = model.IncomingWebhookRequestFromJson(r.Body)
+		payload = r.Body
 	} else {
-		parsedRequest = model.IncomingWebhookRequestFromJson(strings.NewReader(r.FormValue("payload")))
+		payload = strings.NewReader(r.FormValue("payload"))
 	}
 
+	l4g.Debug(func() string {
```

Comment:
Agreed. There are three options:
1. Add a function in mattermost's `utils` package
2. Add a function inside`webhook.go`
3. log4go wrapper

Personally, I think option three is the best, but should be a separate ticket.

The most practical solution in my opinion is option one:
Add a `log.go` file in the `utils` package and add a `func DebugReaderContent(r *io.Reader)`.
I would use a pointer receiver to prevent copying the reader.
Debugging in a closure will not copy the reader, unless loglevel Debug is actually enabled.

Ultimately, it's you choice. What do you prefer?